core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2342,6 +2342,18 @@ private[spark] object Utils extends Logging {
    * Return the initial number of executors for dynamic allocation.
    */
   def getDynamicAllocationInitialExecutors(conf: SparkConf): Int = {
+    if (conf.get(DYN_ALLOCATION_INITIAL_EXECUTORS) < conf.get(DYN_ALLOCATION_MIN_EXECUTORS)) {
+      logWarning(s"${DYN_ALLOCATION_INITIAL_EXECUTORS.key} less than " +
+        s"${DYN_ALLOCATION_MIN_EXECUTORS.key} is invalid, will use " +
+          s"${DYN_ALLOCATION_MIN_EXECUTORS.key} to override.")
```

Comment:
Right but as a user I could interpret that to mean its going to use the min executors for the initial number, which isn't necessarily true.  I'm just worried it will confuse the users.    How about we change the phrasing to be something like: initial less than min is invalid, ignoring its setting, please update your configs.   Similar message for the num executor instances below .

Then after the max calculation we could simply print an info message like: using initial executors = value, max of initial, num executors, min executors.

These warnings are turning into a lot of extra code/text.